File: backend/sdk/php/src/Models/WebhookEndpoint.php
<?php

declare(strict_types=1);

namespace WAAS\Models;

use DateTimeImmutable;

class WebhookEndpoint
{
    public function __construct(
        public readonly string $id,
        public readonly string $tenantId,
        public readonly string $url,
        public readonly bool $isActive,
        public readonly ?RetryConfiguration $retryConfig,
        /** @var array<string, string>|null */
        public readonly ?array $customHeaders,
        public readonly DateTimeImmutable $createdAt,
        public readonly DateTimeImmutable $updatedAt,
        public readonly ?string $secret = null,
    ) {}

    /**
     * @param array<string, mixed> $data
     */
    public static function fromArray(array $data): self
    {
        return new self(
            id: $data['id'],
            tenantId: $data['tenant_id'],
            url: $data['url'],
            isActive: $data['is_active'] ?? true,
            retryConfig: isset($data['retry_config']) 
                ? RetryConfiguration::fromArray($data['retry_config']) 
                : null,
            customHeaders: $data['custom_headers'] ?? null,
            createdAt: new DateTimeImmutable($data['created_at']),
            updatedAt: new DateTimeImmutable($data['updated_at']),
            secret: $data['secret'] ?? null,
        );
    }
}

File: backend/sdk/java/src/main/java/com/waas/sdk/models/WebhookEndpoint.java
package com.waas.sdk.models;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.time.OffsetDateTime;
import java.util.Map;
import java.util.UUID;

/**
 * Webhook endpoint model.
 */
public class WebhookEndpoint {
    private UUID id;
    
    @JsonProperty("tenant_id")
    private UUID tenantId;
    
    private String url;
    
    @JsonProperty("is_active")
    private boolean isActive;
    
    @JsonProperty("retry_config")
    private RetryConfiguration retryConfig;
    
    @JsonProperty("custom_headers")
    private Map<String, String> customHeaders;
    
    @JsonProperty("created_at")
    private OffsetDateTime createdAt;
    
    @JsonProperty("updated_at")
    private OffsetDateTime updatedAt;

    // Secret is only returned on creation
    private String secret;

    public UUID getId() { return id; }
    public void setId(UUID id) { this.id = id; }
    
    public UUID getTenantId() { return tenantId; }
    public void setTenantId(UUID tenantId) { this.tenantId = tenantId; }
    
    public String getUrl() { return url; }
    public void setUrl(String url) { this.url = url; }
    
    public boolean isActive() { return isActive; }
    public void setActive(boolean active) { isActive = active; }
    
    public RetryConfiguration getRetryConfig() { return retryConfig; }
    public void setRetryConfig(RetryConfiguration retryConfig) { this.retryConfig = retryConfig; }
    
    public Map<String, String> getCustomHeaders() { return customHeaders; }
    public void setCustomHeaders(Map<String, String> customHeaders) { this.customHeaders = customHeaders; }
    
    public OffsetDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(OffsetDateTime createdAt) { this.createdAt = createdAt; }
    
    public OffsetDateTime getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(OffsetDateTime updatedAt) { this.updatedAt = updatedAt; }
    
    public String getSecret() { return secret; }
    public void setSecret(String secret) { this.secret = secret; }
}
